mautrix_facebook/analytics.py

```diff
@@ -1,41 +1,44 @@
 from __future__ import annotations
 
+from urllib.parse import urlunparse
 import logging
 
-from yarl import URL
 import aiohttp
 
 from mautrix.util import background_task
 
 from . import user as u
 
 log = logging.getLogger("mau.web.public.analytics")
-segment_url: URL = URL("https://api.segment.io/v1/track")
 http: aiohttp.ClientSession | None = None
-segment_key: str | None = None
-segment_user_id: str | None = None
+analytics_url: str | None = None
+analytics_token: str | None = None
+analytics_user_id: str | None = None
 
 
 async def _track(user: u.User, event: str, properties: dict) -> None:
     await http.post(
-        segment_url,
+        analytics_url,
         json={
-            "userId": segment_user_id or user.mxid,
+            "userId": analytics_user_id or user.mxid,
             "event": event,
             "properties": {"bridge": "facebook", **properties},
         },
-        auth=aiohttp.BasicAuth(login=segment_key, encoding="utf-8"),
+        auth=aiohttp.BasicAuth(login=analytics_token, encoding="utf-8"),
     )
     log.debug(f"Tracked {event}")
 
 
 def track(user: u.User, event: str, properties: dict | None = None):
-    if segment_key:
+    if analytics_token:
         background_task.create(_track(user, event, properties or {}))
 
 
-def init(key, user_id: str | None = None):
-    global segment_key, segment_user_id, http
-    segment_key = key
-    segment_user_id = user_id
+def init(host: str | None, token: str | None, user_id: str | None = None):
+    if not host or not token:
+        return
+    global analytics_url, analytics_token, analytics_user_id, http
+    analytics_url = urlunparse(("https", host, "/v1/track", "", "", ""))
```

Comment:
Using yarl is preferred since that's what aiohttp uses. It should also be stored as a `URL` in the variable like it was before

```suggestion
    analytics_url = URL.build(scheme="https", host=host, path="/v1/track")
```